jx/bdd/terraform.yaml

```diff
@@ -0,0 +1,17 @@
+clusters:
+  - name: bdd-terraform
+    args:
+      - create
+      - terraform
+      - -b
+      - --install-dependencies=true
+      - --skip-login
+      - --organisation-name=bdd
+      - --cloud-provider=gke
+      - --jx-environment=bdd-terraform-$JX_BUILD_NUMBER
```

Comment:
When `step bdd`  is run, it takes the name of the cluster defined above (ie, `bdd-terraform`) and appends `-` and the build number and uses this as the `--cluster-name` parameter for the create operation. As the `create terraform` operation requires the `jx-environment` to match the `cluster-name` to actually install jenkins x in the cluster (see https://github.com/jenkins-x/jx/issues/2372), then these two must match.